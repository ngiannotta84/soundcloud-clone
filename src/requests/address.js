// const address = "https://soundcloud-clone-api.onrender.com";
const address = "http://localhost:4000";

export default address;
