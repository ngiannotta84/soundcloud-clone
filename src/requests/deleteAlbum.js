import { deleteRequest } from "./helpers";

const deleteAlbum = async (id) => {
  const response = await deleteRequest("albums", id);
  return response;
};

export default deleteAlbum;
