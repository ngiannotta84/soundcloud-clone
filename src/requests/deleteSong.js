import { deleteRequest } from "./helpers";

const deleteSong = async (id) => {
  try {
    const response = await deleteRequest("songs", id);
    return response;
  } catch (err) {
    throw new Error(err);
  }
};

export default deleteSong;
