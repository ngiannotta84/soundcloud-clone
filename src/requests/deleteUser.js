import { deleteRequest } from "./helpers";

const deleteUser = async (id, password) => {
  const endpoint = `${id}/${password}`;

  const response = await deleteRequest("users", endpoint);
  return response;
};

export default deleteUser;
