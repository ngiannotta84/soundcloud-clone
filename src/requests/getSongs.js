/* eslint-disable no-console */
import instance from "./instance";

const getSongs = async (obj) => {
  let endpoint = `/songs`;
  const queryArray = [];
  if (obj) {
    if (obj.name) {
      queryArray.push(`name=${obj.name}`);
      if (obj.exact) {
        queryArray.push("exact=true");
      }
    }
    if (obj.limit) {
      queryArray.push(`limit=${obj.limit}`);
    }
    if (queryArray.length > 0) {
      const query = queryArray.join("&");
      endpoint = `${endpoint}?${query}`;
    }
  }

  try {
    const response = await instance.get(endpoint);
    return response.data;
  } catch (err) {
    console.error(err);
    throw new Error(err);
  }
};

export default getSongs;
