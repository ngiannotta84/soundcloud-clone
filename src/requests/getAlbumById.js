/* eslint-disable no-console */
import axios from "axios";
import address from "./address";

const getAlbumById = async (id) => {
  try {
    const response = await axios.get(`${address}/albums/${id}`);
    return response.data;
  } catch (err) {
    console.error(err);
    throw new Error(err);
  }
};

export default getAlbumById;
