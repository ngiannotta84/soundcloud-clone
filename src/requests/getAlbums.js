/* eslint-disable no-console */
import axios from "axios";
import address from "./address";

const getAlbums = async ({ name, exact, limit }) => {
  let endpoint = `${address}/albums`;
  const queryArray = [];
  if (name) {
    queryArray.push(`name=${name}`);
    if (exact) {
      queryArray.push("exact=true");
    }
  }
  if (limit) {
    queryArray.push(`limit=${limit}`);
  }
  if (queryArray.length > 0) {
    const query = queryArray.join("&");
    endpoint = `${endpoint}?${query}`;
  }
  console.log(endpoint);
  try {
    const response = await axios.get(endpoint);
    console.log(response);
    return response.data;
  } catch (err) {
    console.error(err);
    throw new Error(err);
  }
};

export default getAlbums;
