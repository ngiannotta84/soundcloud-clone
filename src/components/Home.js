import React, { useState, useEffect } from "react";
import PropTypes from "prop-types";
import getAlbums from "../requests/getAlbums";
import Album from "./Album";

const Home = ({ handleSetPlaylist }) => {
  const [albums, setAlbums] = useState([]);

  useEffect(() => {
    (async () => {
      try {
        const response = await getAlbums();
        setAlbums(response);
      } catch (err) {
        setAlbums([]);
      }
    })();
  }, []);

  return (
    <div>
      {albums.map((album) => {
        return (
          <Album
            artistName={album.User.name}
            albumName={album.name}
            albumArt={album.url}
            songs={album.Songs}
            key={album.id}
            albumId={album.id}
            handleSetPlaylist={handleSetPlaylist}
          />
        );
      })}
    </div>
  );
};

Home.propTypes = {
  handleSetPlaylist: PropTypes.func.isRequired,
};

export default Home;
