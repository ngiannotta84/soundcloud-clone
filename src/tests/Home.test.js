import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import { BrowserRouter as Router } from "react-router-dom";
import Home from "../components/Home";
import * as getAlbums from "../requests/getAlbums";
import fakeAlbumData from "./testData/fakeAlbumData";

describe("Home", () => {
  const validProps = {
    handleSetPlaylist: jest.fn(),
  };

  beforeEach(() => {
    jest.spyOn(getAlbums, "default").mockResolvedValue(fakeAlbumData);
  });

  test("snapshot", async () => {
    let asFragment;
    await act(() => {
      const view = render(
        <Router>
          <Home handleSetPlaylist={validProps.handleSetPlaylist} />
        </Router>
      );
      asFragment = view.asFragment;
    });

    expect(asFragment()).toMatchSnapshot();
  });

  describe("tests", () => {
    beforeEach(async () => {
      await act(() => {
        render(
          <Router>
            <Home handleSetPlaylist={validProps.handleSetPlaylist} />
          </Router>
        );
      });
    });

    test("renders correctly", async () => {
      await waitFor(() => {
        const images = screen.getAllByRole("img");

        expect(images).toHaveLength(fakeAlbumData.length);
        expect(images[0]).toHaveAttribute(
          "alt",
          `${fakeAlbumData[0].name} cover art`
        );
      });
    });
  });
});
