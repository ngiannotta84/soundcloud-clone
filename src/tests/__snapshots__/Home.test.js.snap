// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="album"
    >
      <div
        class="album__info-container"
      >
        <img
          alt="album1 cover art"
          class="album__cover-art"
          src="art url 1"
        />
        <div>
          <h2
            class="album__artist-name"
          >
            artist name 1
          </h2>
          <h3
            class="album__album-name"
          >
            album1
          </h3>
        </div>
      </div>
      <div>
        <div
          class="album__song"
        >
          <h4
            class="album__song__song-name"
          >
            song1
          </h4>
          <div>
            <button
              class="album__song__add-song"
              type="button"
            >
              Add to queue
            </button>
            <button
              class="album__song__add-song"
              type="button"
            >
              Play next
            </button>
          </div>
        </div>
        <div
          class="album__song"
        >
          <h4
            class="album__song__song-name"
          >
            song2
          </h4>
          <div>
            <button
              class="album__song__add-song"
              type="button"
            >
              Add to queue
            </button>
            <button
              class="album__song__add-song"
              type="button"
            >
              Play next
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="album"
    >
      <div
        class="album__info-container"
      >
        <img
          alt="album2 cover art"
          class="album__cover-art"
          src="art url 2"
        />
        <div>
          <h2
            class="album__artist-name"
          >
            artist name 2
          </h2>
          <h3
            class="album__album-name"
          >
            album2
          </h3>
        </div>
      </div>
      <div>
        <div
          class="album__song"
        >
          <h4
            class="album__song__song-name"
          >
            song4
          </h4>
          <div>
            <button
              class="album__song__add-song"
              type="button"
            >
              Add to queue
            </button>
            <button
              class="album__song__add-song"
              type="button"
            >
              Play next
            </button>
          </div>
        </div>
        <div
          class="album__song"
        >
          <h4
            class="album__song__song-name"
          >
            song3
          </h4>
          <div>
            <button
              class="album__song__add-song"
              type="button"
            >
              Add to queue
            </button>
            <button
              class="album__song__add-song"
              type="button"
            >
              Play next
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
